examples/gaussian_model_selection.py

```diff
@@ -0,0 +1,58 @@
+# A demo of using AIC / BIC for Model Selection
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hmmlearn.hmm import GaussianHMM
+
+# Our model to generate sample data from
+model = GaussianHMM(4, init_params="")
+model.n_features = 4
+model.startprob_ = np.array([1/4., 1/4., 1/4., 1/4.])
+model.transmat_ = np.array([[0.3, 0.4, 0.2, 0.1],
+                            [0.1, 0.2, 0.3, 0.4],
+                            [0.5, 0.2, 0.1, 0.2],
+                            [0.25, 0.25, 0.25, 0.25]])
+model.means_ = np.array([[-2.5], [0], [2.5], [5.]])
+model.covars_ = np.sqrt([[0.25], [0.25], [0.25], [0.25]])
+
+X, _ = model.sample(2000)
+lengths = [200] * 10
+
+# Search over various n_components
+# and examine AIC/BIC/LL
+aic = []
+bic = []
+lls = []
+ns = [2, 3, 4, 5, 6]
+for n in ns:
+    best_ll = None
+    best_model = None
+    # Train a few different models with different
+    # random initializations, saving the one
+    # with the best LL.
+    for i in range(10):
+        h = GaussianHMM(n, n_iter=1000)
+        h.fit(X)
+        score = h.score(X)
+        if not best_ll or best_ll < best_ll:
+            best_ll = score
+            best_model = h
+    aic.append(best_model.aic(X))
+    bic.append(best_model.bic(X))
+    lls.append(best_model.score(X))
+
+# Visualize our results
+fig, ax = plt.subplots()
+ln1 = ax.plot(ns, aic, label="AIC", color="blue", marker="o")
+ln2 = ax.plot(ns, bic, label="BIC", color="green", marker="o")
+ax2 = ax.twinx()
+ln3 = ax2.plot(ns, lls, label="LL", color="orange", marker="o")
+
+lines = ln1 + ln2 + ln3
+labels = [l.get_label() for l in lines]
+ax.legend(lines, labels, loc="best")
```

Comment:
Would write as `ax.legend(handles=ax.lines + ax2.lines)`; no need to explicitly store the lines (`ln1 = ...` above) or compute the labels (`labels = ...`) this way.  Note also that "best" is already the default location.